terraform/errors_test.go

```diff
@@ -0,0 +1,40 @@
+package terraform
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGenerateDebugLogsNotCustomError(t *testing.T) {
+	actual := GenerateDebugLogs(errors.New("Test"))
+	assert.Equal(t, "", actual)
+}
+
+func TestGenerateDebugLogsCustomErrorWithNoErrors(t *testing.T) {
+	actual := GenerateDebugLogs(&CustomError{
+		message: "Test",
+		errors:  []error{},
+	})
+	assert.Equal(t, "", actual)
+}
+
+func TestGenerateDebugLogsCustomErrorWithErrors(t *testing.T) {
+	actual := GenerateDebugLogs(&CustomError{
+		message: "Test",
+		errors: []error{
+			errors.New("Error1"),
+			errors.New("Error2"),
+		},
+	})
+	assert.Equal(t, "\nError1\nError2", actual)
```

Comment:
nit: A few small questions about the logs' content:
Would we perhaps want to have the additional error properties available in these logs somehow? Like the calls stack for example?
Should these logs only include error messages?

Reply:
I don't think they'd be useful and might even end up cluttering the `--debug` output too much. If we eventually want to add the stack trace we can, but for now it's more useful to get the actual error message.

Reply:
Since this is being called from TypeScript using `gopherjs` it's a bit complicated to add additional metadata (the easier way to do this is to return a `map[string]string` like we do here, otherwise the `gopherjs` interface will struggle to parse the nested values)